docs/retention-policy-agent/README.md

```diff
@@ -12,7 +12,75 @@ weight: 2
 The Results Retention Policy Agent removes older Results and their associated Records from the DB.
 
 ## Configuration
-The Results Retention Policy Agent config uses following field
 
-- `runAt`: determines when to run pruning job for DB.
-- `maxRetention`: for how long to keep results and records.
+The Results Retention Policy Agent is configured via the `tekton-results-config-results-retention-policy` ConfigMap.
+
+The following fields are supported:
+
+- `runAt`: Determines when to run the pruning job for the DB. It uses a cron schedule format. The default is `"7 7 * * 7"` (every Sunday at 7:07 AM).
+- `maxRetention`: The default retention period for how long to keep Results and Records when no specific policy matches. This can be a number (e.g., `30`), which is interpreted as days, or a duration string (e.g., `30d`, `24h`). The default is `30d`.
+- `policies`: A list of fine-grained retention policies that allow for more specific control over data retention.
+
+### Fine-Grained Retention Policies
+
+You can define a list of policies to control retention based on various criteria. The `policies` field in the ConfigMap accepts a YAML string containing a list of policy objects. Each policy has a `name`, a `selector`, and a `retention` period.
+
+When the retention job runs, it evaluates a Result against the policies in the order they are defined. The **first policy that matches** the Result will be applied. If no policies match, the default `maxRetention` period is used.
+
+#### Policy Fields:
+- `name`: A descriptive name for the policy.
+- `selector`: Defines the criteria for matching Results. All conditions within a selector must be met for the policy to match.
+  - `matchNamespace`: A list of namespaces. A Result matches if its namespace is in this list.
+  - `matchLabels`: A map where the key is a label name and the value is a list of possible label values. A Result matches if it has a label with the given name and its value is in the list.
+  - `matchAnnotations`: A map where the key is an annotation name and the value is a list of possible annotation values. A Result matches if it has an annotation with the given name and its value is in the list.
+  - `status`: A list of statuses (e.g., `Succeeded`, `Failed`, `Cancelled`). A Result matches if its final status is in this list.
```

Comment:
I guess `Succeeded`, `Failed`, `Cancelled` is an ENUM, right, there aren't other statuses. If that's the case, I'll suggest:
 ```suggestion
  - `status`: A list of statuses (one or more of `Succeeded`, `Failed`, `Cancelled`). A Result matches if its final status is in this list.
```

Reply:
No, there are other statuses. Those are just common one.

Reply:
OK. I have updated the document to highlight this.

Reply:
I also added a link to the docs.